src/test/java/jenkins/plugins/github/api/SmokeTest.java

```diff
@@ -94,15 +81,16 @@ public void given__veryBasicMockGitHub__when__listingManyRepos__then__reposListe
 
             }
             Set<String> actual = new TreeSet<>();
-            for (GHRepository r: openAndConnect(mock).listAllPublicRepositories()) {
+            for (GHRepository r: openAndConnect(connectFunction, mock).listAllPublicRepositories()) {
                 actual.add(r.getFullName());
             }
-            assertThat(actual, is(actual));
+            assertThat(actual, is(expected));
```

Comment:
While I'm certain that `assertThat(actual, is(actual));` was a typo and `assertThat(actual, is(expected));` should be correct, it causes the test to fail. To me it seems like `#listAllPublicRepositories` is not returning all elements for reasons I do not understand.